### PR TITLE
Add release automation

### PR DESCRIPTION
**Pull request by jhkennedy**
This adds a collection of GitHub Actions intended to make releasing as simple as merging to `main`. 

The release notes will be extracted from the `CHANGELOG.md` file and published as part of the release. The assumes:
1. you update the `CHANGELOG.md` before releasing²
2. you label the release PR with the level of changes following semantic versioning conventions

A terse description of the flow is:
1. A PR is opened to `main`
   - `changelog.yml`  will ensure the `CHANGELOG.md` file has been updated
   - `labeled-pr.yml` will ensure the PR is labeled with a label indicating the level of changes 
     (what part of the semantic version tag to bump/increment)
3. Once merged, `tag.yml` will compute the new version tag from the PR label and push a version tag to the repo
4. When a new version tag is pushed, `release.yml` will find the appropriate release notes (changes) in `CHANGELOG.md` based on the version tag, and publish the release on GitHub.

### Footnotes
1. I've copied the release notes from #413, and all previous releases, into the `CHANGELOG.md` file. 
2. I've set the `changelog.yml` workflow to ensure *every* PR includes a changelog. After a release, developers should make a new section for the next release with a proposed version number indicating the level of changes included in their new feature. If the next release section already exists, they should add their changes to it and adjust the proposed number as needed. If there are *no changes that should be communicated to users* (e.g., CI/CD changes), the `bumpless` label can be used to say, "these are changes which do not require a version bump and do not need to be recorded in the changelog," and the check will pass.
---

## TODOs:
- [x] Add the `RELEASE_ACCESS_TOKEN` repository secret
- [x] Add and stylize `major`, `minor`, `patch`, `bumpless`, and `bot` labels
- [x] ~~write a `release-process.md` document describing the release process~~ moved to #418 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have added an explanation of what your changes do and why you'd like us to include them.
- [x] ~~I have written new tests for your core changes, as applicable.~~
- [x] ~~I have successfully ran tests with your changes locally.~~
- [x] My change requires a change to the documentation.
- [x] ~~I have updated the documentation accordingly.~~ Punted to #418

### CODE REVIEW

**Comment by jhkennedy** [comment]
Note: These are the same workflows used in https://github.com/ACCESS-Cloud-Based-InSAR/DockerizedTopsApp

File: .github/workflows/tag.yml
Comment (jhkennedy) [inline .github/workflows/tag.yml:13]: @dbekaert for ACCESS19, we setup a "bot" user which creates the release tag, releases, and syncs `main` back to `dev` if a release branch was used. 

Here I've put you as this repo is under your personal GH account. That means it'll look like you did the tagging/releasing/syncing yourself. For this all to work, you'll need to create a personal access token with write access to this repo and add it as a secret named `RELEASE_ACCESS_TOKEN` to this repo.

Comment (dbekaert) [reply .github/workflows/tag.yml:13]: Ok i will do it in the morning.

**Review by jlmaurer — APPROVED** [review]

File: .github/workflows/tag.yml
Comment (cmarshak) [reply .github/workflows/tag.yml:13]: I would create a jpl org email e.g. `raider@jpl.nasa.gov` and have you, Jeremy, and Joe be the admins on this mailing list. This can then function as your bot - only need to create a github account for this email.

**Review by cmarshak — APPROVED** [review]

**Comment by jhkennedy** [comment]
I've moved 
> write a `release-process.md` document describing the release process

to #418 -- this doesn't need to block the release but should be done soon.

**Comment by dbekaert** [comment]
@jhkennedy i added in the release access token. Can you confirm we are good? feel free to merge if all good.

**Review by dbekaert — APPROVED** [review]
LGTM thanks @jhkennedy

**Comment by jhkennedy** [comment]
> @jhkennedy i added in the release access token. Can you confirm we are good? feel free to merge if all good.

@dbekaert I cannot. I don't have access to any repository settings, and I think *only* you do. If you screenshot the secrets page, I could confirm. But it's CI/CD so "ship it to try it" :rofl:

**Pull request merged**
